sympy/stats/joint_rv.py

```diff
@@ -46,26 +57,49 @@ def value(self):
 
     @property
     def component_count(self):
-        return len(self.distribution.set.args)
+        _set = self.distribution.set
+        if isinstance(_set, ProductSet):
```

Comment:
You could use ternary operator here.
`return len(_set.args) if isinstance(_set, ProductSet) else 1`
Doubt would affect execution time though.